src/main/c/cext/error.c

```diff
@@ -49,3 +49,8 @@ bool rb_warning_category_enabled_p(rb_warning_category_t category) {
 void rb_tr_warn_va_list(const char *fmt, va_list args) {
   RUBY_CEXT_INVOKE("rb_tr_warn", rb_vsprintf(fmt, args));
 }
+
+void rb_error_frozen_object(VALUE frozen_obj) {
```

Comment:
Could you move it to `exception.c`?
In general we don't follow whatever the `.c` file is in CRuby, instead we group by function prefix as documented by a comment at the top of the file.

BTW I'm not sure why we even have this file named as `error.c`, it should be renamed to `warning.c`. Could be done in this PR too while we noticed it.